### PR TITLE
Feat/38 submission approval screen

### PR DESCRIPTION
**Pull request by demariadaniel**

### CODE REVIEW

File: package.json
Comment (demariadaniel) [inline package.json:78]: Moved Prettier rules to package.json as prettierrc was getting ignored, causing unnecessary Git changes
LMK if there's an alternate preferred way to do this, but Platform UI is set up w/ rules in package.json

File: src/app/(post-login)/submission/layout.tsx
Comment (demariadaniel) [inline src/app/(post-login)/submission/layout.tsx:35]: Insure Submission content fits in view window

Comment (ciaranschutte) [reply src/app/(post-login)/submission/layout.tsx:35]: I haven't seen mixing of px and view port units before - interesting!
keep in mind `vw` and `vm` are percents and `px` are absolute

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx
Comment (ciaranschutte) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx:104]: Is this something that needs to be a prop to `ProgressBar` comp? 
seems like it may be a case of "pending approval" progress item is always larger

File: src/global/utils/auth.tsx
Comment (ciaranschutte) [inline src/global/utils/auth.tsx:106]: there are `Suspense` boundary errors on the page and this is the only place in the code where we use suspense boundaries. I wonder if there could be a fix here?

File: src/global/utils/auth.tsx
Comment (ciaranschutte) [inline src/global/utils/auth.tsx:51]: I don't think this returns a string as type
you're on the right track with the react dispatch generics

File: src/global/utils/auth.tsx
Comment (ciaranschutte) [inline src/global/utils/auth.tsx:57]: are these odd placeholder functions needed or can you pass the actual functions?

**Review by ciaranschutte — CHANGES_REQUESTED** [review]

**Comment by ciaranschutte** [comment]
@demariadaniel couple of suggestions. assuming you have the prettier chaos under control from your slack message.

one point. before approval re the context functions. @justincorrigible 2nd opinion please? I may be overlooking something obvious

File: src/global/utils/auth.tsx
Comment (demariadaniel) [reply src/global/utils/auth.tsx:106]: This is from the very early days of setting up the app + Login and definitely worth a revisit esp b/c of the errors

File: package.json
Comment (justincorrigible) [reply package.json]: personally I prefer having separate files for separate concerns...

The manifest file (package.json) tends to get quite busy as it is, in most projects.

In case it helps, `prettier.config.js` works fairly well in Arranger. https://github.com/overture-stack/arranger/blob/develop/prettier.config.js

File: src/app/(post-login)/submission/layout.tsx
Comment (justincorrigible) [reply src/app/(post-login)/submission/layout.tsx:35]: @ciaranschutte the unit mix makes sense if you see height in relation to a "constant" font size (8'd have opted for rem) vs viewport width depending on browser being resized

File: .prettierrc.json
Comment (justincorrigible) [inline .prettierrc.json:6]: The irony of seeing the prettier configs file being reformatted without considering the rules specified therein (tabs)
😂

File: src/app/(post-login)/submission/layout.tsx
Comment (ciaranschutte) [reply src/app/(post-login)/submission/layout.tsx:35]: wouldn't this potentially result in overflow/underflow though? 
if 248px is not 14% (100 - 86) of the view port width  .. that's one col of fixed size, and another col of fixed size that is 86% of viewport. 

are you talking about that idea of setting a base font size and setting everything, including layout dimensions, relative to it?

File: src/app/components/Header.tsx
Comment (justincorrigible) [inline src/app/components/Header.tsx:19]: oof. prettier is making fun of us in this PR. uh?

File: src/app/(post-login)/submission/layout.tsx
Comment (justincorrigible) [reply src/app/(post-login)/submission/layout.tsx:35]: oh, I was strictly focused on the unit mix... the actual values, I have no clue about 🤷‍♂️

Comment (ciaranschutte) [reply src/app/(post-login)/submission/layout.tsx:35]: @demariadaniel  not sure where those numbers came from but could we change this to, otherwise the page gets a little overflowing:
```
 grid-template-columns: ${isSidebarActive
          ? "248px calc(100vw - 248px)"
          : "40px calc(100vw - 40px)"};
```

Comment (justincorrigible) [reply src/app/(post-login)/submission/layout.tsx:35]: Oh. I see what you meant!
Yeah, this suggesrion certainly seems more robust. 

I'd take it a step further by abstracting those constants into reusable variables, that also go into the sidebar itself (so if the number changes at some point, it does it in both places at once)

Comment (ciaranschutte) [reply src/app/(post-login)/submission/layout.tsx:35]: <insert "100" emoji>

File: src/global/utils/auth.tsx
Comment (justincorrigible) [reply src/global/utils/auth.tsx:57]: it seems the code was already there, and the change is just a prettier proble..

while I agree with the suggestion, I think it's way more urgent to amend the PR with Prettier working.

It's difficult to parse (or approve) with all these styling regressions.

Comment (ciaranschutte) [reply src/global/utils/auth.tsx:57]: good point. I didn't check if it was new code or not as my instinct is "see green, comments are fine"

**Comment by justincorrigible** [comment]
I may recommend we get back to reviewing this PR once it's been amended with a working Prettier. if others are ok approving it as is, go ahead, but I don't feel comfortable doing so as is.

File: .prettierrc.json
Comment (justincorrigible) [reply .prettierrc.json:6]: for clarity, comment not intended as anything else but to point out I find this funny.

File: src/app/(post-login)/submission/layout.tsx
Comment (demariadaniel) [reply src/app/(post-login)/submission/layout.tsx]: Great catch @ciaranschutte this did indeed overflow + calc is exactly what I was looking for

For context, I added the `vw` changes to contain the table within the page:

Screenshot 2023-10-27 at 12 11 44 PM
Screenshot 2023-10-27 at 12 11 34 PM

Comment (demariadaniel) [reply src/app/(post-login)/submission/layout.tsx]: Using vw the table fits on the page:
Screenshot 2023-10-27 at 12 10 42 PM

File: package.json
Comment (demariadaniel) [reply package.json]: Fixed in: https://github.com/icgc-argo/rdpc-ui/pull/102

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx
Comment (justincorrigible) [inline src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx:47]: Going to take a wager, and assume the use of FC here is so you can pass children?
I recall ages ago using the FC type was discouraged, because specifying the type explicitly leads to less errors.

you could use `PropsWithChildren<Props>` if children is all you need

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/Header.tsx
Comment (demariadaniel) [reply src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/Header.tsx]: I believe it would (and tested it to be sure), updated!

File: src/global/utils/auth.tsx
Comment (demariadaniel) [reply src/global/utils/auth.tsx]: @ciaranschutte Give this a second look after Prettier revisions. These are just initializing the context, actual functions are passed in the body of AuthProvider. I updated syntax to use `() => {}` instead.

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx
Comment (demariadaniel) [reply src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx:47]: FC<ProgressBarProps> is not my doing. In platform-ui this is `ComponentType<ProgressBarProps>`, which works fine in RDPC as well.
`PropsWithChildren<ProgressBarProps>` does not work for me.

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx
Comment (demariadaniel) [reply src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx]: This is a carryover from platform-ui
It is used here: https://github.com/icgc-argo/platform-ui/blob/develop/components/pages/submission-system/program-dashboard/ProgramWorkspaceStatus/index.tsx#L142

File: src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx
Comment (justincorrigible) [reply src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx:47]: you're right. once again I feel into the "it's green, new code" trap 😂

in either case, at risk of stating the obvious and just to verify a suspicion: the replacement for 
`const X: FC<Props> = props =>...` is `const X = (props: PropsWithChildren<Props>) =>...`, which in turn lets TS infer the return type (a much better way to use TS in React)

is that how you did it?

i.e. ' const X: PropsWithChildren<Props> =...` wouldn't work because X is meant to be a component, not just its props.

Comment (demariadaniel) [reply src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx:47]: OK I see. `const X = (props: PropsWithChildren<Props>) =>` does work.

Comment (justincorrigible) [reply src/app/(post-login)/submission/program/[shortName]/clinical-submission/components/ProgressBar.tsx]: glad to hear! thanks for helping leave these systems better than we found them 🙏

**Comment by ciaranschutte** [comment]
merging based on slack "all PR addressed" comment 👍

**Pull request merged**
